### PR TITLE
added Dockerfile

### PR DESCRIPTION
**Pull request by xinredhat**
1. Added a Dockerfile for running CI jobs
2. Added yamllint.yaml
3. Downgrade golang to 1.21

### CODE REVIEW

File: ci/config/yamllint.yaml
Comment (otaviof) [inline ci/config/yamllint.yaml:1]: Could we move this file to the project root? So, when using `yamllint` outside of CI relies on the same configuration file?

Comment (xinredhat) [reply ci/config/yamllint.yaml:1]: sure, then we move the both files `yamlint.yaml` and `Dockerfile` to the project root, WDYT?

Comment (xinredhat) [reply ci/config/yamllint.yaml:1]: No, I won't move the `Dockerfile` to the project root, as it's only used for CI

Comment (xinredhat) [reply ci/config/yamllint.yaml:1]: fixed

**Review by otaviof — APPROVED** [review]

**Pull request merged**